Core/src/org/sleuthkit/autopsy/datamodel/AttachmentNode.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2017-2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.datamodel;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.logging.Level;
+import javax.swing.Action;
+import org.apache.commons.lang3.StringUtils;
+import org.openide.nodes.Children;
+import org.openide.nodes.Sheet;
+import org.openide.util.Lookup;
+import org.openide.util.NbBundle;
+import org.openide.util.Utilities;
+import org.openide.util.lookup.Lookups;
+import org.sleuthkit.autopsy.actions.AddContentTagAction;
+import org.sleuthkit.autopsy.actions.DeleteFileContentTagAction;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.casemodule.NoCurrentCaseException;
+import org.sleuthkit.autopsy.contentviewers.MessageContentViewer;
+import org.sleuthkit.autopsy.coreutils.ContextMenuExtensionPoint;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import static org.sleuthkit.autopsy.datamodel.FileNode.getIconForFileType;
+import org.sleuthkit.autopsy.directorytree.ExportCSVAction;
+import org.sleuthkit.autopsy.directorytree.ExternalViewerAction;
+import org.sleuthkit.autopsy.directorytree.ExternalViewerShortcutAction;
+import org.sleuthkit.autopsy.directorytree.ExtractAction;
+import org.sleuthkit.autopsy.directorytree.NewWindowViewAction;
+import org.sleuthkit.autopsy.directorytree.ViewContextAction;
+import org.sleuthkit.autopsy.timeline.actions.ViewFileInTimelineAction;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.TskException;
+import org.sleuthkit.datamodel.blackboardutils.Attachment;
+import org.sleuthkit.datamodel.blackboardutils.FileAttachment;
+import org.sleuthkit.datamodel.blackboardutils.URLAttachment;
+
+/**
+ * Node for a message attachment.
+ *
+ */
+public class AttachmentNode extends DisplayableItemNode {
```

Comment:
Does this class have to be public? If so, perhaps we could make it final.

Reply:
AttachmentNode gets used in MessageContentViewers which is in a different package.  I could have left the AttachmentNode in the ContentViewers package but I figured it fits better in the datamodel package with the rest of the nodes.

I'd also hoped to be able to reuse it in MediaViewer, until I realized that the "AttachmentNode" as used in the MediaViewer was in spirit - a "AttachmentThumbnailNode".

I certainly can be final though.

Reply:
Sounds good, let's make it final.